### PR TITLE
OpenAPI controller - Bootstrap UI

### PR DESCRIPTION
**Pull request by Tymek**
## About the changes

Add `src/lib/routes/admin-api/bootstrap-ui.ts` to API validation and documentation.

### CODE REVIEW

**Comment by Tymek** [comment]
[OpenAI: src/lib/routes/admin-api/bootstrap-controller.ts](https://trello.com/c/Y3PtIBME/1339-openai-src-lib-routes-admin-api-bootstrap-controllerts)

File: src/lib/openapi/index.ts [hunk 1]
```diff
@@ -1,118 +1,120 @@
 import { OpenAPIV3 } from 'openapi-types';
+import { addonParameterSchema } from './spec/addon-parameter-schema';
```
Comment (Tymek) [inline src/lib/openapi/index.ts:2]: Maybe eslint import auto-sorting is a good option for the future :)

File: src/lib/routes/admin-api/bootstrap-ui.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import createStores from '../../../test/fixtures/store';
+import getApp from '../../app';
+import { createServices } from '../../services';
+const uiConfig = {
+    headerBackground: 'red',
+    slogan: 'hello',
+};
+
+async function getSetup() {
+    const base = `/random${Math.round(Math.random() * 1000)}`;
```
[the remark this answers is not in the document]
Comment (Tymek) [reply src/lib/routes/admin-api/bootstrap-ui.test.ts:13]: @sonatype-lift ignore

File: src/lib/openapi/spec/bootstrap-ui-schema.test.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
+import { validateSchema } from '../validate';
+
+test('bootstrapUiSchema', () => {
+    const data = {
```
Comment (olav) [inline src/lib/openapi/spec/bootstrap-ui-schema.test.ts:4]: Should use the generated type here to test that as well:

```
const data: BootstrapUiSchema = {
```

File: src/lib/openapi/spec/bootstrap-ui-schema.test.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                featureCount: 10,
+                memberCount: 3,
+                updatedAt: '2022-06-28T17:33:53.963Z',
+            },
+        ],
+    };
+
+    expect(
+        validateSchema('#/components/schemas/bootstrapUiSchema', data),
+    ).toBeUndefined();
```
Comment (olav) [inline src/lib/openapi/spec/bootstrap-ui-schema.test.ts:151]: Could add another test to ensure that the validation at least returns some error:

```
expect(
    validateSchema('#/components/schemas/bootstrapUiSchema', {}),
).not.toBeUndefined();
```

File: src/lib/routes/admin-api/bootstrap-ui.test.ts (hunk 2 above)
Comment (olav) [inline src/lib/routes/admin-api/bootstrap-ui.test.ts:13]: Could use `randomId()` instead.

File: src/lib/routes/admin-api/bootstrap-ui.ts [hunk 5]
```diff
@@ -71,15 +87,30 @@ class BootstrapController extends Controller {
...
+        this.logger = config.getLogger('routes/admin-api/bootstrap-ui.ts');
+        this.route({
+            method: 'get',
+            path: '',
+            handler: this.bootstrap,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['other'],
+                    operationId: 'bootstrapUiSchema',
```
Comment (olav) [inline src/lib/routes/admin-api/bootstrap-ui.ts:101]: Would give the handler and operationId the same name, maybe `getBootstrapUi` or `getUiBootstrapData`.

**Review by olav — CHANGES_REQUESTED** [review]

**Review by olav — APPROVED** [review]

**Pull request merged**